flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/ActiveResourceManagerFactory.java

```diff
@@ -81,4 +83,14 @@ private static Configuration createActiveResourceManagerConfiguration(Configurat
 		ClusterInformation clusterInformation,
 		@Nullable String webInterfaceUrl,
 		ResourceManagerMetricGroup resourceManagerMetricGroup) throws Exception;
+
+	protected WorkerResourceSpec createDefaultWorkerResourceSpec(Configuration configuration) {
```

Comment:
I'm wondering whether this should really be a method belonging to the `ActiveResourceManagerFactory`. If I look at the body, then it could perfectly be a static method. No need to be method of `ActiveResourceManagerFactory` since it does not access any state.

Reply:
It calls to the abstract method `getDefaultCpus`, which is currently overridden differently by different `ActiveResourceManagerFactory` implementations.